lib/HLSL/WaveSensitivityAnalysis.cpp

```diff
@@ -42,13 +44,22 @@ using namespace std;
 
 namespace hlsl {
 
+// WaveSensitivityAnalysis is created to validate Gradient operations.
+// Gradient operations require all neighbor lanes to be active when calculate,
+// compiler will enable lanes to meet this requirement. If a wave operation
+// contributed gradient operation, it will get unexpected result because the
+// active lanes is modified.
+// To avoid unexpected result, validation will fail if gradient operations
+// dependent on wave-sensitive data or control flow.
```

Comment:
"calculate," -> "calculated;"
"contributed gradient operation" -> "contributed to a gradient operation"
"active lanes is modified" -> "active lanes are modified"
"dependent on" -> "are dependent on"